### PR TITLE
Fix post config verification without flags.

### LINKED ISSUE #19408: tls-verify in daemon config file does not enable TLS
TLS is only ever enabled if `"tls": true` is specified in the daemon config.
With the CLI flags, `--tlsverify` enables tls, but not so with the config file.

### PR DESCRIPTION
**Pull request by calavera**
- Set the daemon log level to what's set in the configuration.
- Enable TLS when TLSVerify is enabled.

Fixes #19408

Signed-off-by: David Calavera david.calavera@gmail.com

### CODE REVIEW

File: docker/daemon.go [hunk 1]
```diff
@@ -360,5 +360,13 @@ func loadDaemonCliConfig(config *daemon.Config, daemonFlags *flag.FlagSet, commo
 		}
 	}
 
+	// ensure that TLS is enabled when TLSVerify is enabled
+	if config.TLSVerify && !config.TLS {
```
Comment (tiborvass) [inline docker/daemon.go]: you could remove `&& !config.TLS`

**Comment by tiborvass** [comment]
@thaJeztah @calavera what about the `cmd.IsSet()` equivalent in https://github.com/docker/docker/blob/master/docker/common.go#L72 ? If tlsverify is set to false, explicitly on the command line, tls is set to true. Super weird but that's how it is.

**Comment by calavera** [comment]
> If tlsverify is set to false, explicitly on the command line, tls is set to true

We generate a conflict if tls-verify is set to false in the flags and true in the configuration.

tls == true and tls-verify == false is a valid configuration, fwiw.

**Comment by tiborvass** [comment]
@calavera I wasn't talking about the command vs config file conflict, but the config-file equivalent of the command line flags.
IOW if i set tlsverify to false in the config file, shouldn't it set tls to true, just like `--tlsverify=false` sets tls to true?

**Comment by thaJeztah** [comment]
I agree with @tiborvass, it's a weird one, but better be consistently weird

**Comment by calavera** [comment]
I fixed that case too.

**Comment by cpuguy83** [comment]
daemon/config.go:118:1: receiver name c should be consistent with previous receiver name config for Config

**Comment by calavera** [comment]
> daemon/config.go:118:1: receiver name c should be consistent with previous receiver name config for Config

fixed, thanks!

**Comment by thaJeztah** [comment]
ping @tiborvass @cpuguy83 PTAL :)

File: docker/daemon.go [hunk 2]
```diff
@@ -360,5 +360,13 @@ func loadDaemonCliConfig(config *daemon.Config, daemonFlags *flag.FlagSet, commo
 		}
 	}
 
+	// ensure that TLS is enabled when TLSVerify is enabled
+	if config.TLSVerify || config.IsValueSet("tls-verify") {
```
Comment (tiborvass) [inline docker/daemon.go]: no need for `config.TLSVerify ||` unless i'm mistaken

Comment (tiborvass) [inline docker/daemon.go]: ping @calavera wdyt?

Comment (calavera) [inline docker/daemon.go]: fixed

File: docker/daemon.go [hunk 3]
```diff
@@ -360,5 +360,14 @@ func loadDaemonCliConfig(config *daemon.Config, daemonFlags *flag.FlagSet, commo
 		}
 	}
 
+	// ensure that TLS is enabled when TLSVerify is set
+	// in the configuration file
```
Comment (thaJeztah) [inline docker/daemon.go]: Maybe we should add the same note here to explain we enable TLS both for tis-verify = true AND false; people may be confused and try to fix it in future. taken from `common.go`;

```
// Regardless of whether the user sets it to true or false, if they
// specify --tlsverify at all then we need to turn on tls
```

Comment (calavera) [inline docker/daemon.go]: done

**Comment by cpuguy83** [comment]
LGTM... unless you want to fix this in this PR too... invalid config options are currently silently ignored.
This could be really bad particularly with `tls-verify`, if someone does something like `tlsverify` instead.

**Comment by tiborvass** [comment]
LGTM

**Comment by thaJeztah** [comment]
@cpuguy83 thanks, I've opened an issue for it https://github.com/docker/docker/issues/19515

**Pull request merged**